gempython/tools/amc_user_functions_xhal.py

```diff
@@ -254,8 +264,17 @@ def configureVFAT3DacMonitorMulti(self, dacSelect, ohMask=0xFFF):
         dacSelect - An integer defining the monitored register.
                     See VFAT3 Manual GLB_CFG_CTR_4 for details.
         ohMask - Mask which defines which OH's to query; 12 bit number where
-                 having a 1 in the N^th bit means to query the N^th optohybrid
+                 having a 1 in the N^th bit means to query the N^th optohybrid.
+                 If None will be determined automatically using HwAMC::getOHMask()
         """
+        
+        # Automatically determine ohMask if not provided
+        if ohMask is None:
+            ohMask = self.getOHMask()
+
+        # Are any optohybrids available?
+        if (not (bin(ohMask).count("1") > 0) ):
+            raise NoUnmaskedOHException("{0}HwAMC::configureVFAT3DacMonitorMulti(): there are no unmasked optohybrids{1}".format(colors.RED,colors.ENDC),os.EX_SOFTWARE)
```

Comment:
The following code snippet (ll270-277) appears in multiple places unchanged. Thus, the code should be refactored and the above snippet should be factored out as a method, e.g.
```python
def determineOHMask(ohMask=None):
    if ohMask is None:
             ohMask = self.getOHMask()

          # Are any optohybrids available?
         if (not (bin(ohMask).count("1") > 0) ):
             raise NoUnmaskedOHException("{0}HwAMC::configureVFAT3DacMonitorMulti(): there are no unmasked optohybrids{1}".format(colors.RED,colors.ENDC),os.EX_SOFTWARE)
    return ohMask
```

Reply:
sure I'll think of a solution.

Reply:
resolved